bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java

```diff
@@ -268,12 +267,12 @@ public void removeLedgerMetadata(final long ledgerId, final Version version,
             cb.operationComplete(BKException.Code.MetadataVersionException, (Void)null);
             return;
         } else if (Version.ANY != version) {
-            if (!(version instanceof ZkVersion)) {
+            if (!(version instanceof LongVersion)) {
                 LOG.info("Not an instance of ZKVersion: {}", ledgerId);
                 cb.operationComplete(BKException.Code.MetadataVersionException, (Void)null);
                 return;
             } else {
-                znodeVersion = ((ZkVersion)version).getZnodeVersion();
+                znodeVersion = (int) ((LongVersion) version).getLongVersion();
```

Comment:
the (int) cast sounds a bit weird, but for me it is OK, it is not possible to do it in other way without deep changes.
this patch is all a "rename"